LukeMason33/Rotten-Tomatillos
# Feature router

### Is this a fix or a feature?

Feature.

### What is the change/fix?

This feature updates the conditional rendering to instead use react-router. It also refactors the `SingleMovieView` function component into its own component, giving it its own state and fetch. This allows users to send single move page links and have those links render correctly.

### Where should the reviewer start?

Review changes made to `render()` on `App.js`. Note how the id is being passed to the `SingleMovieView` component from `match.params.id`. Review the `SingleMovieView` component, and the fetch request being run on `componentDidMount()`.


### How should this be tested?

In browser, click the details button on a single movie. Ensure that the URL is being updated to show the movie title and id. Copy the single movie URL, open a new tab, and paste the URL into the address bar to ensure that the single movie page is able to load and display independent from main page navigation.
